axiom/tests/test_log_events.c

```diff
@@ -142,33 +161,125 @@ static void test_events_sample(void) {
+  tlib_pass_if_true("retrived log element from vector OK", pass,
+                    "expected TRUE");
+  tlib_pass_if_str_equal("LOG_MESSAGE_3 log event retained", LOG_MESSAGE_3,
+                         ((nr_log_event_t*)test_e)->message);
+  tlib_pass_if_int_equal("Priority 50 log event retained", 50,
+                         ((nr_log_event_t*)test_e)->priority);
+  pass = nr_vector_get_element(vector, 1, &test_e);
+  tlib_pass_if_true("retrived log element from vector OK", pass,
+                    "expected TRUE");
+  tlib_pass_if_str_equal("LOG_MESSAGE_0 log event retained", LOG_MESSAGE_0,
+                         ((nr_log_event_t*)test_e)->message);
+  tlib_pass_if_int_equal("Priority 100 log event retained", 100,
+                         ((nr_log_event_t*)test_e)->priority);
+
+  nr_vector_destroy(&vector);
+
+  // Test: Add another priority 50 log event and it should remove oldest
+  e = create_sample_event(LOG_MESSAGE_4);
+  nr_log_event_set_priority(e, 50);
+  nr_log_event_set_timestamp(e, (LOG_TIMESTAMP + 100) * NR_TIME_DIVISOR_MS);
+  event_dropped = nr_log_events_add_event(events, e);
+  tlib_pass_if_true("final event caused drop", event_dropped,
+                    "nr_log_events_add_event: got [%d], want [%d]",
+                    event_dropped, true);
+
+  // verify two highest priority events were retained
+  // convert log event pool to a vector and test
+  vector = nr_vector_create(10, NULL, NULL);
+  nr_log_events_to_vector(events, vector);
+  pass = nr_vector_get_element(vector, 0, &test_e);
+  tlib_pass_if_true("retrived log element from vector OK", pass,
+                    "expected TRUE");
+  tlib_pass_if_str_equal("LOG_MESSAGE_4 log event retained", LOG_MESSAGE_4,
+                         ((nr_log_event_t*)test_e)->message);
+  tlib_pass_if_int_equal("Priority 50 log event retained", 50,
+                         ((nr_log_event_t*)test_e)->priority);
+  pass = nr_vector_get_element(vector, 1, &test_e);
+  tlib_pass_if_true("retrived log element from vector OK", pass,
+                    "expected TRUE");
+  tlib_pass_if_str_equal("LOG_MESSAGE_0 log event retained", LOG_MESSAGE_0,
+                         ((nr_log_event_t*)test_e)->message);
+  tlib_pass_if_int_equal("Priority 100 log event retained", 100,
+                         ((nr_log_event_t*)test_e)->priority);
+
+  nr_vector_destroy(&vector);
+
   nr_log_events_destroy(&events);
   tlib_pass_if_null("events destroyed", events);
 }
 
+static void test_log_event_comparator(void) {
+  nr_log_event_t log0 = {.message = "LOG0", .priority = 30, .timestamp = 0};
+  nr_log_event_t log1 = {.message = "LOG1", .priority = 20, .timestamp = 0};
+  nr_log_event_t log2 = {.message = "LOG2", .priority = 10, .timestamp = 0};
+  nr_log_event_t log3 = {.message = "LOG3", .priority = 0, .timestamp = 0};
+  nr_log_event_t log4 = {.message = "LOG4", .priority = 0, .timestamp = 100};
+
+  nr_vector_t* events = nr_vector_create(12, NULL, NULL);
+
```

Comment:
```suggestion
  // clang-format off
```

Reply:
I don't think this will be reformatted I kept the line lengths short enough.